src/mlpack/tests/gan_test.cpp

```diff
@@ -201,21 +202,22 @@ BOOST_AUTO_TEST_CASE(GANMNISTTest)
   generator.Add<ReLULayer<> >();
   generator.Add<BilinearInterpolation<> >(28, 28, 56, 56, noiseDim / 4);
   generator.Add<Convolution<> >(noiseDim / 4, 1, 3, 3, 2, 2, 1, 1, 56, 56);
-  generator.Add<TanHLayer<> >();
+  generator.Add<SigmoidLayer<> >();
```

Comment:
I don't see why the change here is necessary, it is [recommended](https://github.com/soumith/ganhacks) practice to use `TanH` layer as the last layer of the `Generator`.

Reply:
In the O'Reilly test they use a sigmoid layer at the end in the generator. So, just to be consistent.
But I can revert this change. let me know.

Reply:
Please do.

Reply:
>  it is recommended practice to use TanH layer as the last layer of the Generator, since we need to keep the images between -1 and 1.

I am a bit confused about this. The training data we use is unnormalised. (it has large positive values and zeros)
Whereas in the O'Reilly test their images are normalized to [0, 1] and they have sigmoid layer at the end of the generator. Do you think we need to normalize the images to [-1, 1] and use the tanH layer at the end?

Reply:
Read my comment [here](https://github.com/mlpack/mlpack/pull/1724#discussion_r263302250). This test in its current form is only there to serve as a measure of logical correctness. You can add normalization for the images, it will slow down your computations a bit, but you can do without them as well. It's the layers that matter the most.